server/controllers/ArticleController.js

```diff
@@ -270,6 +277,71 @@ class ArticleController {
         return next(unknownActionError);
     }
   }
+
+  /**
+    * @description - This method adds a new report.
+    * @param {object} req - The request object with report body and type.
+    * @param {object} res - The response object that is returned as json.
+    * @returns {object} - The object with message.
+    * @memberOf UserController
+    */
+  static reportArticle(req, res) {
+    const { reportBody, type } = req.body;
+    const userId = req.userData.id;
+    const { articleId } = req.params;
+
+    Article.findByPk(articleId)
+      .then((article) => {
+        ReportType.findOne({
+          where: {
+            title: type
+          }
+        })
+          .then((result) => {
+            const typeId = result.id;
```

Comment:
What happens if a user enters a `type` value that doesn't exist in the `ReportType` table? I think you should add an if statement here to catch that, something like:
```
if (!result) {
    // return error message that type doesn't exist
}
```
Good work implementing this feature, everything else looks good!

Reply:
This is fixed now @Darthrighteous. Good find.